src/current-user.js

```diff
@@ -246,14 +261,41 @@ export class CurrentUser {
     })
     return this.joinRoom(roomId)
       .then(room => this.roomSubscriptions[roomId].connect().then(() => room))
-      .catch(err => this.logger.error(
-        `error subscribing to room ${roomId}`,
-        err
-      ))
+      .catch(err => {
+        this.logger.warn(`error subscribing to room ${roomId}:`, err)
+        throw err
+      })
+  }
+
+  fetchAttachment = url => {
+    return this.filesInstance.tokenProvider.fetchToken()
+      .then(token => sendRawRequest({
+        method: 'GET',
+        headers: { Authorization: `Bearer ${token}` },
+        url
+      }))
+      .then(pipe(JSON.parse, parseFetchedAttachment))
+      .catch(err => {
+        this.logger.warn(`error fetching attachment:`, err)
+        throw err
+      })
   }
 
   /* internal */
 
+  uploadDataAttachment = (roomId, { file, name }) => {
+    // TODO some validation on allowed file names?
+    // TODO polyfill FormData?
```

Comment:
We'll have to do this at some point (or use lower level methods), but the existing JS sdk uses `FormData` as well, and the current goal is to match functionality with what already exists asap so that we can get ths released.